src/pages/ChattingRoom.tsx

```diff
@@ -0,0 +1,209 @@
+import { useState, useEffect, useRef } from "react";
+import { useRecoilState, useRecoilValue } from "recoil";
+import { currentUsersState } from "../recoil/recoil";
+import styled from "styled-components";
+import { chatInterface, userInterface } from "../interfaces/interface";
+import usersData from "../json/usersData.json";
+import chatsData from "../json/chatsData.json";
+import Alert from "../components/Alert";
+import SmallButton from "../components/SmallButton";
+import Profile from "../components/Profile";
+import LeftChat from "../components/LeftChat";
+import RightChat from "../components/RightChat";
+
+const ChattingRoom = () => {
+  const currentUsers = useRecoilValue<userInterface[]>(currentUsersState);
+  const [typingUser, setTypingUser] = useState<userInterface>(
+    usersData.users[0]
+  );
+  const nonTypingUser = currentUsers.filter(
+    (user) => user.userId !== typingUser.userId
+  )[0];
+  const [chatList, setChatList] = useState<chatInterface[]>(
+    chatsData.chattings[0].chatList
+  ); // TODO - recoil로 변경
+  // const [chatList, setChatList] = useRecoilState(currentChatListState);
+
+  const [inputText, setInputText] = useState<string>("");
+  const isTyping: boolean = inputText.trim() ? true : false;
+
+  const [isVisibleAlert, setIsVisibleAlert] = useState<boolean>(false);
+
+  const textareaRef = useRef<HTMLTextAreaElement>(null);
+  const bottomDivRef = useRef<HTMLDivElement>(null);
+
+  const handleInputChange = (e: React.ChangeEvent<HTMLTextAreaElement>) => {
+    setInputText(e.target.value);
+  };
+
+  const handleEnterKeyUp = (e: React.KeyboardEvent<HTMLTextAreaElement>) => {
+    if (!e.shiftKey && e.key === "Enter") {
+      handleSubmit();
+    }
+  };
+
+  const handleSubmit = (e?: React.FormEvent<HTMLFormElement>) => {
```

Comment:
이부분은 수진님 리뷰에도 남겨놓은 부분인데, 잘 모르겠는 부분이 있어서 여쭤보려구 남깁니다! 저번 발표때 `handleSubmit`에 event를 ?.(optional chaining)을 사용해서 예외처리를 해주었다고 했었던 것 같은데, `handleEnterKeyUp`에서 `handleSubmit`함수를 실행할 때 event가 넘어오지 않는 경우가 있나요~??

Reply:
`handleSubmit` 에서 KeyboardEvent를 사용하지 않기 때문에 넘겨주지 않았고, 그래서 인수를 전달하지 않는 경우 예외처리를 위해 optional change를 사용했습니다.
만약 `handleSubmit` 에서 KeyboardEvent 를 사용한다면 타입을 두 개 설정해주면 될 것 같아요!
```suggestion
  const handleSubmit = (e?: React.FormEvent<HTMLFormElement>| React.KeyboardEvent<HTMLTextAreaElement>) => {
```